source/find_getmore_killcursors_commands.rst

```diff
@@ -365,18 +365,54 @@ More detailed information about the interaction of the **secondaryOk** with **OP
 Behavior of Limit, skip and batchSize
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-The new **find** command has different semantics to the existing 3.0 and earlier **OP_QUERY** wire protocol message. The **limit** field is a hard limit on the total number of documents returned by the cursor no matter what **batchSize** is provided.
+The new **find** command has different semantics to the existing 3.0 and earlier
+**OP_QUERY** wire protocol message. The **limit** field is a hard limit on the
+total number of documents returned by the cursor no matter what **batchSize** is
+provided. This includes other limiting operations, such as the **$limit**
+aggregation pipeline stage. This differs from existing **OP_QUERY** behavior
+where there is no server-side concept of limit and where the driver **MUST**
+keep track of the limit on the client side and **MUST** send a
+**OP_KILL_CURSORS** wire protocol message when it limit is reached.
+
+When setting the **batchSize** on the **find** and **getMore** commands the
+value of **batchSize** **MUST** be based on the cursor limit calculations
+specified in the `CRUD`_ specification.
+
+Note that with 5.0, the server-side handling of cursors with a limit has
+changed. Before 5.0, some cursors were automatically closed when the limit was
+reached (e.g. when running **find** with **limit**), and the reply document did
+not include a cursor ID (i.e. ``cursor.id`` was ``0``). Starting with 5.0, all
+cursor-producing operations will return a cursor ID if the end of the batch
+being returned lines up with the limit on the cursor. In this case, drivers
+**MUST** ensure the cursor is closed on the server. To do this, drivers **MUST**
+use one of the options provided below:
+
+* send an additional **getMore** command with **batchSize** 1. Since the limit
```

Comment:
I think this options is only included because the C driver currently does this.  But would we want any driver that didn't already do this to choose this option?  I would think not, as option 2 is much simpler.  If there's agreement on this point, I think we should consider just removing this option from the PR, and just call the C driver out of spec until DRIVERS-1448.

Reply:
The reason I offered both is because they achieve the same result.

If we're looking at DRIVERS-1448, then the result of not tracking the limit on the cursor means that an additional `getMore` will be required, as the only indication that we've reached the limit will be the absence of a cursor ID in the reply document. With that in mind, I would continue to suggest both options, meaning that every driver conforms to the spec until changes may become necessary with DRIVERS-1448.

Reply:
Discussed in person, and agreed that some changes will be made here to simplify the requirements.

Reply:
Updated to remove the detailed suggestions. The spec now asks drivers to ensure the cursor is closed by exhausting it or using `killCursors`.